sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,297 @@ def to_rotation_matrix(self, v=None):
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.vector()
+        0 + 1*i + 1*j + 1*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.vector()
+        0 + 8*i + 13*j + 12*k
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d)
+
+    def axis(self):
+        """
+        Returns the axis that is the versor of the vector part of the quaternion.
+
+        Returns
+        =======
+
+        The axis of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.axis()
+        0 + sqrt(3)/3*i + sqrt(3)/3*j + sqrt(3)/3*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.axis()
+        0 + 8*sqrt(377)/377*i + sqrt(377)/29*j + 12*sqrt(377)/377*k
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d).normalize()
+
+    def is_pure(self):
+        """
+        Returns true if the quaternion is pure else returns false.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.is_pure()
+        False
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.is_pure()
+        True
+
+        """
+
+        return self.a == 0 and not (self.a == self.b == self.c == self.d)
+
+    def angle(self):
+        """
+        Returns the angle of the quaternion measured in the real-axis plane.
+
+        Returns
+        =======
+
+        The angle of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 4, 4, 4)
+        >>> q.angle()
+        atan(4*sqrt(3))
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.angle()
+        pi/2
+
+        """
+
+        return atan2(Quaternion(0, self.b, self.c, self.d).norm(), self.a)
+
+    def coplanar(self, other):
+        """
+        Returns True if the plane of the two quaternions are coplanar else returns False.
+
+        Parameters
+        ==========
+
+        other : a quaternion
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 4, 4, 4)
+        >>> q1 = Quaternion(2, 8, 8, 8)
+        >>> q.coplanar(q1)
+        True
+
+        >>> q1 = Quaternion(2, 8, 13, 12)
+        >>> q.coplanar(q1)
+        False
+
+        """
+        if not self.is_pure() and self.a == 0 or not other.is_pure() and other.a == 0:
+            raise ValueError('Any of the provided quaternions must not be 0')
+
+        aq = self.axis()
+        ar = other.axis()
+
+        return (aq == ar or aq == -ar)
+
+    def ternary_coplanar(self, q1, q2):
+        """
+        Returns True if the axis of the pure quaternions seen as 3D vectors
+        self, q1, and q2 are coplanar else returns False.
```

Comment:
Where does `ternary_coplanar` come from? Would it make more sense to have a class method like `Quaternion.coplanar` that can work for any number of quaternions?